routes/compat/wakatime/v1/leaders.go

```diff
@@ -0,0 +1,82 @@
+package v1
+
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/muety/wakapi/helpers"
+	"github.com/muety/wakapi/middlewares"
+	"github.com/muety/wakapi/models"
+	"github.com/muety/wakapi/utils"
+	"net/http"
+
+	conf "github.com/muety/wakapi/config"
+	v1 "github.com/muety/wakapi/models/compat/wakatime/v1"
+	"github.com/muety/wakapi/services"
+)
+
+type LeadersHandler struct {
+	config          *conf.Config
+	userSrvc        services.IUserService
+	leaderboardSrvc services.ILeaderboardService
+}
+
+func NewLeadersHandler(userService services.IUserService, leaderboardService services.ILeaderboardService) *LeadersHandler {
+	return &LeadersHandler{
+		userSrvc:        userService,
+		leaderboardSrvc: leaderboardService,
+		config:          conf.Get(),
+	}
+}
+
+func (h *LeadersHandler) RegisterRoutes(router chi.Router) {
+	router.Group(func(r chi.Router) {
+		r.Use(middlewares.NewAuthenticateMiddleware(h.userSrvc).Handler)
+		r.Get("/compat/wakatime/v1/leaders", h.Get)
+	})
+}
+
+// @Summary List of users ranked by coding activity in descending order.
+// @Description Mimics https://wakatime.com/developers#leaders
+// @ID get-wakatime-leaders
+// @Tags wakatime
+// @Produce json
+// @Security ApiKeyAuth
+// @Success 200 {object} v1.LeaderboardViewModel
+// @Router /compat/wakatime/v1/leaders [get]
+func (h *LeadersHandler) Get(w http.ResponseWriter, r *http.Request) {
+	wakapiUser, err := h.userSrvc.GetUserById("current")
+	if err != nil {
+		conf.Log().Request(r).Error("error while fetching user - %v", err)
+		helpers.RespondJSON(w, r, http.StatusInternalServerError, nil)
+		return
+	}
+	pageParams := utils.ParsePageParamsWithDefault(r, 1, 100)
+
+	/*
+		Not implemented query params:
+			language - String - optional - Filter leaders by a specific language.
+			is_hireable - Boolean - optional - Filter leaders by the hireable badge.
+			country_code - String - optional - Filter leaders by two-character country code.
+			page - Integer - optional - Page number of leaderboard. If authenticated, defaults to the page containing the currently authenticated user.
```

Comment:
Would you mind implementing the pagination? You can probably take inspiration from the leaderboard route.

Reply:
https://github.com/Subtixx/wakapi/blob/b2c7042e183cdeca5ebcdbe6a960369bb7e26a41/routes/compat/wakatime/v1/leaders.go#L52-L63

This should do pagination shouldn't it?